### PR TITLE
Cumulative counters need to be used for dimensional metrics

### PR DESCRIPTION
**Pull request by tsheasha**
Derivative function does not respect dimensions

### CODE REVIEW

**Comment by poros** [comment]
+1

File: src/diamond/collectors/cpu/cpu.py [hunk 1]
```diff
@@ -209,57 +202,37 @@ def cpu_delta_time(interval):
...
-                self.publish(metric_name + '.idle',
-                             self.derivative(metric_name + '.idle',
-                                             total_time.idle,
-                                             self.MAX_VALUES['idle'])
-                             / cpu_count)
+                    self.publish_cumulative_counter(metric_name + '.nice',
+                                                 total_time.nice / cpu_count)
+                self.publish_cumulative_counter(metric_name + '.system',
+                                             total_time.system / cpu_count)
+                self.publish_cumulative_counter(metric_name + '.idle',
```
Comment (gnufied) [inline src/diamond/collectors/cpu/cpu.py]: Where does this `total_time` comes from? I don't see it defined or declared anywhere.

Comment (tsheasha) [inline src/diamond/collectors/cpu/cpu.py]: i just pushed a patch for that, but good catch anyways

Comment (gnufied) [inline src/diamond/collectors/cpu/cpu.py]: Also, I think total time reporting should be outside this loop:

```for i in range(0, len(cpu_time)):```

Comment (tsheasha) [inline src/diamond/collectors/cpu/cpu.py]: +1

**Comment by gnufied** [comment]
:shipit:  after green build.

**Pull request merged**